torch/csrc/jit/codegen/cuda/scheduler/pointwise_utils.cpp

```diff
@@ -0,0 +1,99 @@
+#include <torch/csrc/jit/codegen/cuda/scheduler/pointwise_utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace pointwise_utils {
+
+DomainMap::DomainMap(Fusion* fusion)
+    : fusion_(fusion), ca_map_(ComputeAtMap(fusion)) {
+  view_tvs_ = scheduler_utils::getViewTVs(fusion);
+}
+
+bool DomainMap::areMapped(IterDomain* id1, IterDomain* id2) {
+  return ca_map_.areMapped(id1, id2, IdMappingMode::EXACT);
+}
+
+// Determine if all iterDomains are mapped between input and output tvs
```

Comment:
nit: it seems it only checks if all of the input domains are mapped to the output domains, no matter if there exists some output domains that are not mapped with any of the input domains. Is it intentional?

Reply:
I am not sure. This code is copy-pasted from the pointwise scheduler unchanged. My understanding is that for the pointwise scheduler, this is intentional. Because this function is used to find an output that is eligible to be a reference tensor, in such a case, the output having more ID shouldn't be a problem for us. If my understanding is correct, then:
1. I believe this function needs a better name
2. I am using this function incorrectly in this PR because I am allowing both inputs and outputs to be reference tensors, and I should not use this function to test if an input tensor is a reference tensor.

Reply:
Renamed in https://github.com/csarofeen/pytorch/pull/1867